graph.py

```diff
@@ -0,0 +1,57 @@
+import random
+
+
+class Graph:
+    def __init__(self):
+        """
+        graph variable is a dictionary of lists - graph is implemented as adjacency lists
+        """
+        self.graph = dict()
+        self.nodes = set()
+
+    def _topological_sort(self, root, visited, result):
+        visited.add(root)
+
+        for neighbour in self.graph[root]:
+            if neighbour not in visited:
+                result, visited = self._topological_sort(neighbour, visited, result)
+
+        result.append(root)
+
+        return result, visited
+
+    def _get_random_node(self, nodes):
+        return random.choice(nodes)
+
+    def start_topological_sort(self, root=None):
+        visited = set()
+        result = list()
+        nodes = list(self.graph.keys())
+        print(nodes)
+
+        if not nodes:
+            return []
+
+        if root is None:
```

Comment:
It seems weird/odd to me to start with a root node set to None, but I recognise you probably wrote this condition for checking for cases where this might occur?